resources/vocabulary.md

```diff
@@ -93,8 +93,18 @@ sum_of_nums_under(4) // 10
 sum_of_nums_under(6) // 21
 ```
 
+## Process and Tools
+- IDE: Integrated Development Environment - a piece of software that you write code in, typically with helpful features like tab completion, compilation, and debugging
+- Debugging: figuring out how to make your code work as intended when it isn’t - there’s a story about a moth in a mainframe, but the idea is that something isn’t behaving as expected. There are actual programs that step through the execution of your code and allow you to determine how things are happening, or there’s the low-tech approach of just printing out a bunch of stuff.
+- Algorithm: a step by step sequence of events to accomplish a goal, like a recipe.
+- Pseudocode: a helpful step in the development process of writing out what you plan to do in regular language before coding it.
+- Framework: A scaffold that you can build applications around. Includes reusable code that provides a set of basic, generic functionality for you to build upon. A common example is MVC (Model-View-Controller) frameworks for building web apps, such as Django.
+- Library/ Package: A set of resources that someone else has already written and provided so that you can make use of them rather than having to write that code yourself. An example would be the string library in C.
+- REPL: Stands for Read-Eval-Print-Loop. Simple programming environment that acts as a language shell and lets us write and test code without having to instal and configure a bunch of prerequisites.
+- CLI: Command Line Interface. Interacting with your computer via text commands in the shell.
+
+
 ### TODO(lpercival): add other vocabulary
 - Specific words from descriptions of common concepts? 
```

Comment:
Hmm.. I totally don't remember what I meant here. Maybe just delete this part now?